File: leetcode/1800-maximum-ascending-subarray-sum/maxAscendingSum.ts
function maxAscendingSum(nums: number[]): number {
  let sum = 0;
  let maxSum = 0;
  let preNum = 0;

  for (const num of nums) {
    if (num <= preNum) {
      sum = num;
      preNum = num;
      continue;
    }

    preNum = num;
    sum = sum + num;
    maxSum = Math.max(maxSum, sum);
  }

  maxSum = Math.max(maxSum, sum);
  return maxSum;
}

export default maxAscendingSum;

File: leetcode/1800-maximum-ascending-subarray-sum/maxAscendingSum.go
package leetcode

func maxAscendingSum(nums []int) int {
	var (
		maxSum = 0
		sum    = 0
		preNum = 0
	)

	for _, num := range nums {
		if num <= preNum {
			sum = num
			preNum = num
			continue
		}

		preNum = num
		sum = sum + num
		maxSum = max(maxSum, sum)
	}

	maxSum = max(maxSum, sum)

	return maxSum
}

func max(num1 int, num2 int) int {
	if num1 > num2 {
		return num1
	}

	return num2
}
